File: Leetcode第232题.cpp
class MyQueue {
public:
    stack<int> in;
    stack<int> out;
    MyQueue() {

    }
    
    void push(int x) {
        in.push(x);
    }
    
    int pop() {
        if(out.size()==0){
            while(in.size()!=0)
            {
                out.push(in.top());
                in.pop();
            }
        }
        int head=out.top();
        out.pop();
        return head;
    }
    
    int peek() {
        int res=this->pop();
        out.push(res);
        return res;
    }
    
    bool empty() {
        return in.empty()&&out.empty();
    }
};

File: Leetcode第232题.py
class MyQueue(object):
    def __init__(self):
        self.s=[]
    def push(self, x):
        self.s.append(x)
    def pop(self):
        res=self.s[0]
        self.s.remove(self.s[0])
        return res
    def peek(self):
        return self.s[0]
    def empty(self):
        if(len(self.s)==0):
            return True
        else:
            return False
